broker/models/Encounter.js

```diff
@@ -9,50 +9,50 @@ Encounter {
 */
 const CLASS_SYSTEM = "http://terminology.hl7.org/CodeSystem/v3-ActCode";
 exports.toFHIR = function (encounter) {
-  let resource = {
-    resourceType: "Encounter",
-    status: encounter.status,
-    class: {
-      system: CLASS_SYSTEM,
-      code: "FLD",
-      display: "FLD",
-    },
-    subject: {
-      reference: encounter.patient,
-    },
-    appointment: [
-      {
-        reference: encounter.appointment,
-      },
-    ],
-    period: { start: new Date().toISOString() },
-    location: [
-      {
-        location: {
-          reference: encounter.location,
+    const resource = {
+        resourceType: "Encounter",
+        status: encounter.status,
+        class: {
+            system: CLASS_SYSTEM,
+            code: "FLD",
+            display: "FLD"
         },
-      },
-    ],
-    serviceProvider: {
-      reference: encounter.serviceProvider,
-    },
-  };
+        appointment: [
+            {
+                reference: encounter.appointment
+            }
+        ],
+        period: { start: new Date().toISOString() },
+        location: [
+            {
+                location: {
+                    reference: encounter.location
+                }
+            }
+        ],
+        subject: {
+            reference: encounter.subject ?? encounter.patient
+        },
+        serviceProvider: {
+            reference: encounter.serviceProvider
+        }
+    };
 
-  return resource;
+    return resource;
 };
 
 exports.toModel = (encounter) => {
-  let model = {
-    resourceType: encounter.resourceType,
-    id: encounter.id,
-    status: encounter.status,
-    class: encounter.class.code,
-    patient: encounter.subject.reference,
-    appointment: encounter.appointment[0].reference,
-    location: encounter.location[0].location.reference,
-    start: encounter.period.start,
-    end: encounter.period.end,
-  };
+    const model = {
+        resourceType: encounter.resourceType,
+        id: encounter.id,
+        status: encounter.status,
+        class: encounter.class.code,
+        subject: encounter.subject.reference,
```

Comment:
What format does the front end provide? I tried looking at it but I couldn't figure it out. If you could help me figure this out, this would also obviate the need for the ?? operator.

Reply:
What format does the front end provide? I tried looking at it but I couldn't figure it out. If you could help me figure this out, this would also obviate the need for the ?? operator.

Reply:
The format is (should be) the comment in the respective models/resource.js file

Reply:
I changed it to patient.